KifuSF/StatusPagerTabStripViewController.swift

```diff
@@ -9,119 +9,90 @@
-        separatorLine.translatesAutoresizingMaskIntoConstraints = false
-        
-        barView.autoSetDimension(.height, toSize: 44)
-        barView.autoPin(toTopLayoutGuideOf: self, withInset: 0)
-        barView.autoPinEdge(toSuperviewEdge: .leading)
-        barView.autoPinEdge(toSuperviewEdge: .trailing)
-        barView.autoPinEdge(.bottom, to: .top, of: separatorLine)
-        
-        separatorLine.autoSetDimension(.height, toSize: 0.5)
-        separatorLine.autoPinEdge(toSuperviewEdge: .leading)
-        separatorLine.autoPinEdge(toSuperviewEdge: .trailing)
-        separatorLine.autoPinEdge(.bottom, to: .top, of: ownContainerView)
-        
-        ownContainerView.autoPinEdge(toSuperviewEdge: .leading)
-        ownContainerView.autoPinEdge(toSuperviewEdge: .trailing)
-        ownContainerView.autoPin(toBottomLayoutGuideOf: self, withInset: 0)
-        
-        self.buttonBarView = barView
-        self.containerView = ownContainerView
-    }
-    
-    private func setUpTopBar() {
-        settings.style.buttonBarBackgroundColor = UIColor.Pallete.White
-        settings.style.buttonBarItemBackgroundColor = UIColor.Pallete.White
-        settings.style.selectedBarBackgroundColor = UIColor.Pallete.Green
-        settings.style.buttonBarItemFont = UIFont.systemFont(ofSize: 14)
-        settings.style.selectedBarHeight = 2.0
-        settings.style.buttonBarMinimumLineSpacing = 0
-        settings.style.buttonBarItemTitleColor = .black
-        settings.style.buttonBarItemsShouldFillAvailableWidth = true
-        settings.style.buttonBarItemFont = UIFont.preferredFont(forTextStyle: .subheadline)
-        
-        settings.style.buttonBarLeftContentInset = 0
-        settings.style.buttonBarRightContentInset = 0
-        
-        changeCurrentIndexProgressive = { (oldCell: ButtonBarViewCell?, newCell: ButtonBarViewCell?, progressPercentage: CGFloat, changeCurrentIndex: Bool, animated: Bool) -> Void in
-            guard changeCurrentIndex == true else { return }
-            oldCell?.label.textColor = UIColor.Text.Headline
-            newCell?.label.textColor = UIColor.Pallete.Green
-        }
-    }
+}
 
-    private func setUpFirebase() {
-        guard
-            let deliveryVc = controllerArray[0] as? DeliveryModularTableViewController,
-            let donationVc = controllerArray[1] as? DonationModularTableViewController else {
-                fatalError("either the indexes of these elements in this array are out of order or the class types have changed for one of the view controllers")
-        }
-        
-        DonationService.observeCurrentDelivery { (delivery) in
-            deliveryVc.delivery = delivery
-        }
-        
-        DonationService.observeCurrentDonation { (donation) in
-            donationVc.donation = donation
-        }
+extension StatusPagerTabStripViewController: FlaggingContentItems {
+    var flaggableItems: [FlaggedContentType] {
```

Comment:
I have just noticed that theoretically, we have the functionality to flag your own donation post.

Reply:
I have added new logic for handling logic.